### PR TITLE
go-runner: add a REGISTRY substitution to allow registry selection

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
same work we did for kube-cross: https://github.com/kubernetes/release/pull/1943 we are applying for go-runner

k/test-infra PR to update the jobs: https://github.com/kubernetes/test-infra/pull/21316

/assign @justaugustus @saschagrunert @hasheddan 

tracking issue: https://github.com/kubernetes/release/issues/1850

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: images/Makefile.common-image [hunk 1]
```diff
@@ -27,10 +27,16 @@ ARCHS = $(patsubst linux/%,%,$(PLATFORMS))
...
+check-env:
+ifndef REGISTRY
+	$(error REGISTRY is undefined, please set to registry you want to push)
+endif
+
 # build with buildx
 # https://github.com/docker/buildx/issues/59
 .PHONY: container
-container: init-docker-buildx
+container: check-env init-docker-buildx
```
Comment (hasheddan) [inline images/Makefile.common-image:39]: @cpanato are there any images that don't use `REGISTRY` but do use `Makefile.common-image`?

Comment (cpanato) [reply images/Makefile.common-image:39]: right now only go-runner reference the file ` Makefile.common-image`

**Comment by cpanato** [comment]
`pull-release-image-go-runner` succeed 🎉

**Comment by justaugustus** [comment]
> `pull-release-image-go-runner` succeed 🎉

That's how we do it! Great work, @cpanato! 🎉 
/lgtm
/approve

**Pull request merged**